network-area-diagram/src/main/java/com/powsybl/nad/svg/CustomLabelProvider.java

```diff
@@ -0,0 +1,107 @@
+/**
+ * Copyright (c) 2025, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+package com.powsybl.nad.svg;
+
+import com.powsybl.nad.model.*;
+import com.powsybl.nad.utils.svg.SvgUtils;
+
+import java.util.*;
+
+/**
+ * @author Christian Biasuzzi {@literal <christian.biasuzzi at soft.it>}
+ */
+public class CustomLabelProvider implements LabelProvider {
+    final Map<String, CustomBranchLabels> branchLabels;
+    final Map<String, CustomThreeWtLabels> threeWtLabels;
+    final Map<String, String> busDescriptions;
+    final Map<String, List<String>> vlDescriptions;
+    final Map<String, List<String>> vlDetails;
+
+    public record CustomBranchLabels(String side1, String middle, String side2, EdgeInfo.Direction arrow1, EdgeInfo.Direction arrow2) {
```

Comment:
As you're within the CustomLabelProvider, you can remove the Custom prefix (same for the two other records).
```suggestion
    public record BranchLabels(String side1, String middle, String side2, EdgeInfo.Direction arrow1, EdgeInfo.Direction arrow2) {
```